presence-firestore/functions/index.js

```diff
@@ -47,11 +51,20 @@ exports.onUserStatusChanged = functions.database.ref('/status/{uid}').onUpdate(
       if (status.last_changed > eventStatus.last_changed) {
         return null;
       }
-
       // Otherwise, we convert the last_changed field to a Date
       eventStatus.last_changed = new Date(eventStatus.last_changed);
 
-      // ... and write it to Firestore.
-      return userStatusFirestoreRef.set(eventStatus);
+      if(status.state === 'offline') {
+        // ... and write it to Firestore.
+        await userStatusFirestoreRef.delete();
+
+        if ((await userDeviceCollectionRef.get()).empty) {
```

Comment:
`async/await` syntax targets ES2016+, which is inconsistent with the rest of the file. Looking for whether ES5 is the target, or ES2016+